common/predictive-text/worker/model-compositor.ts

```diff
@@ -49,6 +49,7 @@ class ModelCompositor {
     let postContext = models.applyTransform(inputTransform, context);
     let keepOptionText = this.lexicalModel.wordbreak(postContext);
     let keepOption: Suggestion = null;
+    let punctuation = this.lexicalModel.punctuation || DEFAULT_PUNCTUATION;
```

Comment:
Should this be an extend from `DEFAULT_PUNCTUATION` rather than an either or? Allows unset parameters to drop through. I see the compiler does this now, but it would be better here I think because older versions of lexical models may not include newer punctuation properties (i.e. we cannot rely on models being recompiled for a newer version of Keyman).

Reply:
My thoughts are that it should be handled during each model's initialization.  As it stands, `punctuation` is being passed as an argument to each model, right?  The model can handle setting the defaults then, and if nothing is specified, it can call a `common` function (yet-to-be-defined) that auto-populates any missing fields of the object or simply returns the default settings wholesale if none are specified in the `options` initialization object.

That said, as the definitions currently stand, either all of the punctuation or none of it must be specified - none of `LexicalModelPunctuation`'s fields are optional.  Is that intended?

Reply:
> My thoughts are that it should be handled during each model's initialization. As it stands, `punctuation` is being passed as an argument to each model, right? The model can handle setting the defaults then, and if nothing is specified, it can call a `common` function (yet-to-be-defined) that auto-populates any missing fields of the object or simply returns the default settings wholesale if none are specified in the `options` initialization object.

Sure, I'll get this to work on instantiation. Though I think the model compositor will do any filling out of details.

> That said, as the definitions currently stand, either all of the punctuation or none of it must be specified - none of `LexicalModelPunctuation`'s fields are optional. Is that intended?

This is intended in this early version, though allowing all fields to be optional is definitely preferred.